pkg/replication/replication.go

```diff
@@ -111,8 +114,49 @@ func (repl *Replicator) Enact(store IntentStore, serviceChecker ServiceChecker,
 					// node is out of date, update the store with the new manifest.
 					// this will almost definitely result in duplicate Sets in the
 					// preparer, which is expected to ignore duplicate updates.
+					intentPath := kp.IntentPath(node.Name, repl.Manifest.ID())
+					existingManifest, dur, err := store.Pod(intentPath)
+					if err != nil && err != pods.NoCurrentManifest {
+						repl.Logger.WithFields(logrus.Fields{
+							"err":      err,
+							"node":     node.Name,
+							"duration": dur,
+						}).Errorln("Could not read current manifest")
+						continue
+					}
+
+					// If manifest has an AllowedDeployers list, we need to check it against the signature.
+					if err != pods.NoCurrentManifest && existingManifest.AllowedDeployers != nil {
```

Comment:
This is a substantially complex new addition to this method and deserves its own modularization + tests.